blockchain/skipchain/block.go

```diff
@@ -176,37 +212,133 @@ func (fl ForwardLink) computeHash() ([]byte, error) {
 	return h.Sum(nil), nil
 }
 
+// Chain is an ordered list of skipblocks.
+type Chain []SkipBlock
+
+// GetProof returns a verifiable proof to the latest block of the chain.
+func (c Chain) GetProof() Proof {
+	chain := c[:len(c)-1]
+
+	proof := Proof{
+		ForwardLinks: make([]ForwardLink, len(chain)),
+	}
+
+	for i, block := range chain {
+		proof.ForwardLinks[i] = block.ForwardLinks[0]
+	}
+
+	return proof
+}
+
+// Proof is a chain of forward links to a specific block so that the chain
+// can be followed and verified.
+type Proof struct {
+	GenesisBlock SkipBlock
+	ForwardLinks []ForwardLink
+}
+
+// Pack implements the Packable interface to encode the proof into a protobuf
+// message.
+func (p Proof) Pack() (proto.Message, error) {
+	proof := &ProofProto{
+		ForwardLinks: make([]*ForwardLinkProto, len(p.ForwardLinks)),
+	}
+
+	for i, link := range p.ForwardLinks {
+		packed, err := link.Pack()
+		if err != nil {
+			return nil, xerrors.Errorf("couldn't pack the forward link: %v", err)
+		}
+
+		proof.ForwardLinks[i] = packed.(*ForwardLinkProto)
```

Comment:
The encoding is implemented in this module and always the same type of protobuf message so it can be assumed as it cannot be different. (And unit test will catch if someone changes it.)

Reply:
I will do it anyway as it makes sense after our discussion on the consensus interface.